tests/testthat/test-deploymentTarget.R

```diff
@@ -300,3 +472,79 @@ test_that("findUnique always returns unique name", {
   expect_equal(findUnique("x", c("x", "x-1")), "x-2")
   expect_equal(findUnique("x", c("x", "x-1", "x-2")), "x-3")
 })
+
+test_that("createDeploymentFromApplication promotes fields", {
+  expect_equal(createDeploymentFromApplication(
+    application = list(
+      id = "1",
+      name = "app-name",
+      title = "app-title",
+      owner_username = "alice.username"
+    ),
+    accountDetails = list(
+      name = "alice",
+      server = "example.com"
+    )
+  ), list(
+    name = "app-name",
+    title = "app-title",
+    envVars = NULL,
+    appId = "1",
+    username = "alice.username",
+    account = "alice",
+    server = "example.com",
+    version = deploymentRecordVersion
+  ))
+})
+
+test_that("updateDeployment updates fields", {
+  expect_equal(updateDeployment(
+    list(
+      name = "app-name",
+      title = "app-title",
+      envVars = NULL,
+      appId = "1",
+      username = "alice.username",
+      account = "alice",
+      server = "example.com",
+      version = deploymentRecordVersion
+    ),
+    appTitle = "updated-title",
+    envVars = c("VAR-NAME")
+  ), list(
+    name = "app-name",
+    title = "updated-title",
+    envVars = c("VAR-NAME"),
+    appId = "1",
+    username = "alice.username",
+    account = "alice",
+    server = "example.com",
+    version = deploymentRecordVersion
+  ))
+})
+
+test_that("updateDeployment ignores NULL updates", {
```

Comment:
Adding these tests let me see that we were mixing records from disk which used `name` and records created on-the-fly which used `appName`. We are now consistent.